src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsrestrictedpatientsapi/config/HmppsRestrictedPatientsApiExceptionHandler.kt

```diff
@@ -35,6 +35,21 @@ class HmppsRestrictedPatientsApiExceptionHandler {
       )
   }
 
+  @ExceptionHandler(BadRequestException::class)
+  fun handleValidationException(e: BadRequestException): ResponseEntity<ErrorResponse> {
+    log.info("Bad Request exception: {}, {}", e.errorCode, e.message)
+    return ResponseEntity
+      .status(BAD_REQUEST)
+      .body(
+        ErrorResponse(
+          status = BAD_REQUEST,
+          errorCode = e.errorCode,
+          developerMessage = "Validation failure: ${e.message}",
+          userMessage = e.message
+        )
+      )
+  }
```

Comment:
Added the custom BadRequestException to hold an errorCode that we can send back to the caller helping them differentiate.